File: common/src/main/java/cn/breadnicecat/lovemod/item/ModItems.java
package cn.breadnicecat.lovemod.item;

import cn.breadnicecat.lovemod.LoveMod;
import cn.breadnicecat.lovemod.item.items.DivorceAgreement;
import cn.breadnicecat.lovemod.item.items.EngagementRing;
import cn.breadnicecat.lovemod.item.items.WeddingRing;
import dev.architectury.registry.registries.DeferredRegister;
import dev.architectury.registry.registries.DeferredSupplier;
import dev.architectury.registry.registries.RegistrySupplier;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.item.Item;

import static cn.breadnicecat.lovemod.LoveMod.LOGGER;
import static cn.breadnicecat.lovemod.utils.Utils.prefix;


/**
 * Created in 2024/8/27 11:19
 * Project: LoveModFurina
 *
 * @author <a href="https://github.com/Bread-Nicecat">Bread_NiceCat</a>
 * <p>
 *
 * <p>
 **/
public class ModItems {
	static {
		LOGGER.info("Loading Items");
	}
	
	public static DeferredRegister<Item> register = DeferredRegister.create(LoveMod.MOD_ID, Registries.ITEM);
	
	public static RegistrySupplier<EngagementRing> ENGAGEMENT_RING = register.register(prefix("engagement_ring"), EngagementRing::new);
	public static DeferredSupplier<WeddingRing> WEDDING_RING = register.register(prefix("wedding_ring"), WeddingRing::new);
	public static DeferredSupplier<DivorceAgreement> DIVORCE_AGREEMENT = register.register(prefix("divorce_agreement"), DivorceAgreement::new);
}

File: common/src/main/kotlin/cn/breadnicecat/lovemod/item/ModItems.kt
package cn.breadnicecat.lovemod.item

import cn.breadnicecat.lovemod.LoveMod
import cn.breadnicecat.lovemod.LoveMod.LOGGER
import cn.breadnicecat.lovemod.Utils.prefix
import cn.breadnicecat.lovemod.item.items.Ring
import cn.breadnicecat.lovemod.item.items.RingType
import dev.architectury.registry.registries.DeferredRegister
import dev.architectury.registry.registries.DeferredSupplier
import net.minecraft.core.registries.Registries
import net.minecraft.world.item.Item

/**
 * Created in 2024/8/27 11:19
 * Project: LoveModFurina
 *
 * @author <a href="https://github.com/Bread-Nicecat">Bread_NiceCat</a>
 * <p>
 *
 * <p>
 **/
object ModItems {
    init {
        LOGGER.info("Loading Items")
    }

    val register: DeferredRegister<Item> = DeferredRegister.create(LoveMod.MOD_ID, Registries.ITEM);

    val MALE_RING: DeferredSupplier<Ring> = register.register(prefix("male_ring")) { Ring(RingType.MALE) }
    val FEMALE_RING: DeferredSupplier<Ring> = register.register(prefix("female_ring")) { Ring(RingType.FEMALE) }

}